--- pvemanagerlib.8.4.0.js	2026-01-08 00:24:46.998846298 +0000
+++ pvemanagerlib.8.4.0_new.js	2026-01-08 00:33:59.155713951 +0000
@@ -9519,6 +9519,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['quantastor', 'QuantaStor-API'],
 	['istgt', 'istgt'],
 	['iet', 'IET'],
 	['LIO', 'LIO'],
@@ -61144,6 +61145,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+		isQuantaStor: false,
 	    hasWriteCacheOption: true,
 	},
     },
@@ -61160,6 +61162,12 @@
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
 	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+		vm.set('isQuantaStor', newVal === 'quantastor');
+		if (newVal !== 'quantastor') {
+			me.lookupReference('qs_apiv4_host_field').setValue('');
+			me.lookupReference('qs_user_field').setValue('');
+			me.lookupReference('qs_password_field').setValue('');
+		}
 	},
     },
 
@@ -61193,7 +61201,7 @@
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'pool',
 		value: '',
 		fieldLabel: gettext('Pool'),
@@ -61207,7 +61215,7 @@
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'target',
 		value: '',
 		fieldLabel: gettext('Target'),
@@ -61218,9 +61226,19 @@
 		name: 'comstar_tg',
 		value: '',
 		fieldLabel: gettext('Target group'),
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: { hidden: '{!isComstar}' },
 		allowBlank: true,
 	    },
+		{
+		    xtype: 'textfield',
+		    name: 'qs_user',
+		    reference: 'qs_user_field',
+		    inputId: 'qs_user_field',
+		    value: '',
+		    allowBlank: false,
+		    fieldLabel: gettext('QS Username'),
+		    bind: { hidden: '{!isQuantaStor}' },
+		}
 	];
 
 	me.column2 = [
@@ -61234,18 +61252,20 @@
 	    {
 		xtype: 'proxmoxcheckbox',
 		name: 'sparse',
-		checked: false,
+		checked: true,
 		uncheckedValue: 0,
 		fieldLabel: gettext('Thin provision'),
 	    },
 	    {
-		xtype: 'proxmoxcheckbox',
-		name: 'writecache',
-		checked: true,
-		bind: me.isCreate ? { disabled: '{!hasWriteCacheOption}' } : { hidden: '{!hasWriteCacheOption}' },
-		uncheckedValue: 0,
-		fieldLabel: gettext('Write cache'),
-	    },
+        xtype: 'proxmoxcheckbox',
+        name: 'writecache',
+        checked: true,
+        bind: me.isCreate
+            ? { disabled: '{!hasWriteCacheOption}' }
+            : { hidden: '{!hasWriteCacheOption}' },
+        uncheckedValue: 0,
+        fieldLabel: gettext('Write cache'),
+        },
 	    {
 		xtype: me.isCreate ? 'textfield' : 'displayfield',
 		name: 'comstar_hg',
@@ -61262,6 +61282,31 @@
 		allowBlank: false,
 		fieldLabel: gettext('Target portal group'),
 	    },
+		{
+	        xtype: 'proxmoxtextfield',
+	        name: 'qs_apiv4_host',
+	        reference: 'qs_apiv4_host_field',
+	        value: '',
+	        editable: true,
+	        allowBlank: true,
+	        emptyText: Proxmox.Utils.noneText,
+	        bind: { hidden: '{!isQuantaStor}' },
+	        fieldLabel: gettext('QS Host'),
+	        allowBlank: true
+	    },
+	    {
+	        xtype: 'proxmoxtextfield',
+	        name: 'qs_password',
+	        reference: 'qs_password_field',
+	        inputType: me.isCreate ? '' : 'password',
+	        value: '',
+	        editable: true,
+	        deleteEmpty: true,
+	        allowBlank: false,
+	        emptyText: Proxmox.Utils.noneText,
+	        bind: { hidden: '{!isQuantaStor}' },
+	        fieldLabel: gettext('QS Password')
+	    },
 	];
 
 	me.callParent();
