--- pvemanagerlib.8.4.0.js	2025-11-21 21:12:06.177467044 +0000
+++ pvemanagerlib.8.4.0_new.js	2025-11-21 21:12:06.185467058 +0000
@@ -9519,6 +9519,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
 	['comstar', 'Comstar'],
+	['quantastor', 'QuantaStor-API'],
 	['istgt', 'istgt'],
 	['iet', 'IET'],
 	['LIO', 'LIO'],
@@ -61144,6 +61145,7 @@
 	data: {
 	    isLIO: false,
 	    isComstar: true,
+		isQuantaStor: false,
 	    hasWriteCacheOption: true,
 	},
     },
@@ -61160,6 +61162,15 @@
 	    vm.set('isLIO', newVal === 'LIO');
 	    vm.set('isComstar', newVal === 'comstar');
 	    vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+		vm.set('isQuantaStor', newVal === 'quantastor');
+		vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt')
+		if (newVal !== 'quantastor') {
+			me.lookupReference('qs_use_ssl_field').setValue(true);
+			me.lookupReference('qs_apiv4_host_field').setValue('');
+			me.lookupReference('qs_user_field').setValue('');
+			me.lookupReference('qs_password_field').setValue('');
+			me.lookupReference('qs_confirmpw_field').setValue('');
+		}
 	},
     },
 
@@ -61177,6 +61188,7 @@
     },
 
     setValues: function(values) {
+	values.qs_confirmpw = values.qs_password;
 	values.writecache = values.nowritecache ? 0 : 1;
 	this.callParent([values]);
     },
@@ -61193,7 +61205,7 @@
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'pool',
 		value: '',
 		fieldLabel: gettext('Pool'),
@@ -61207,7 +61219,7 @@
 		allowBlank: false,
 	    },
 	    {
-		xtype: me.isCreate ? 'textfield' : 'displayfield',
+		xtype: 'textfield',
 		name: 'target',
 		value: '',
 		fieldLabel: gettext('Target'),
@@ -61218,9 +61230,29 @@
 		name: 'comstar_tg',
 		value: '',
 		fieldLabel: gettext('Target group'),
-		bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+		bind: { hidden: '{!isComstar}' },
 		allowBlank: true,
 	    },
+		{
+		xtype: 'proxmoxcheckbox',
+		name: 'qs_use_ssl',
+		reference: 'qs_use_ssl_field',
+		inputId: 'qs_use_ssl_field',
+		checked: true,
+		bind: { hidden: '{!isQuantaStor}' },
+		uncheckedValue: 0,
+		fieldLabel: gettext('API use SSL')
+	    },
+	    {
+		xtype: 'textfield',
+		name: 'qs_user',
+		reference: 'qs_user_field',
+		inputId: 'qs_user_field',
+		value: '',
+		allowBlank: false,
+		fieldLabel: gettext('API Username'),
+		bind: { hidden: '{!isQuantaStor}' },
+		}
 	];
 
 	me.column2 = [
@@ -61262,6 +61294,57 @@
 		allowBlank: false,
 		fieldLabel: gettext('Target portal group'),
 	    },
+		{
+		xtype: 'proxmoxtextfield',
+		name: 'qs_apiv4_host',
+		reference: 'qs_apiv4_host_field',
+		value: '',
+		editable: true,
+		allowBlank: true,
+		emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isQuantaStor}' },
+		fieldLabel: gettext('API IPv4 Host'),
+		allowBlank: true
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'qs_password',
+		reference: 'qs_password_field',
+		inputType: me.isCreate ? '' : 'password',
+		value: '',
+		editable: true,
+		deleteEmpty: true,
+		allowBlank: false,
+		emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isQuantaStor}' },
+		fieldLabel: gettext('API Password'),
+		change: function(f, value) {
+		    if (f.rendered) {
+			f.up().down('field[name=qs_confirmpw]').validate();
+		    }
+		}
+	    },
+	    {
+		xtype: 'proxmoxtextfield',
+		name: 'qs_confirmpw',
+		reference: 'qs_confirmpw_field',
+		inputType: me.isCreate ? '' : 'password',
+		value: '',
+		editable: true,
+		deleteEmpty: true,
+		allowBlank: false,
+		submitValue: false,
+		emptyText: Proxmox.Utils.noneText,
+		bind: { hidden: '{!isQuantaStor}' },
+		fieldLabel: gettext('Confirm Password'),
+		validator: function(value) {
+		    var pw = this.up().down('field[name=qs_password]').getValue();
+		    if (pw !== value) {
+			return "Passwords do not match!";
+		    }
+		    return true;
+		}
+	    },
 	];
 
 	me.callParent();
