--- pvemanagerlib.9.1.1.js	2026-01-08 00:37:48.524075551 +0000
+++ pvemanagerlib.9.1.1_new.js	2026-01-08 00:47:16.220978277 +0000
@@ -9840,6 +9840,7 @@
     alias: ['widget.pveiScsiProviderSelector'],
     comboItems: [
         ['comstar', 'Comstar'],
+        ['quantastor', 'QuantaStor-API'],
         ['istgt', 'istgt'],
         ['iet', 'IET'],
         ['LIO', 'LIO'],
@@ -66980,6 +66981,7 @@
         data: {
             isLIO: false,
             isComstar: true,
+            isQuantaStor: false,
             hasWriteCacheOption: true,
         },
     },
@@ -66995,7 +66997,13 @@
             var vm = this.getViewModel();
             vm.set('isLIO', newVal === 'LIO');
             vm.set('isComstar', newVal === 'comstar');
+            vm.set('isQuantaStor', newVal === 'quantastor');
             vm.set('hasWriteCacheOption', newVal === 'comstar' || newVal === 'istgt');
+            if (newVal !== 'quantastor') {
+		    	me.lookupReference('qs_apiv4_host_field').setValue('');
+		    	me.lookupReference('qs_user_field').setValue('');
+		    	me.lookupReference('qs_password_field').setValue('');
+		    }
         },
     },
 
@@ -67029,7 +67037,7 @@
                 allowBlank: false,
             },
             {
-                xtype: me.isCreate ? 'textfield' : 'displayfield',
+                xtype: 'textfield',
                 name: 'pool',
                 value: '',
                 fieldLabel: gettext('Pool'),
@@ -67043,7 +67051,7 @@
                 allowBlank: false,
             },
             {
-                xtype: me.isCreate ? 'textfield' : 'displayfield',
+                xtype: 'textfield',
                 name: 'target',
                 value: '',
                 fieldLabel: gettext('Target'),
@@ -67054,9 +67062,19 @@
                 name: 'comstar_tg',
                 value: '',
                 fieldLabel: gettext('Target group'),
-                bind: me.isCreate ? { disabled: '{!isComstar}' } : { hidden: '{!isComstar}' },
+                bind: { hidden: '{!isComstar}' },
                 allowBlank: true,
             },
+	        {
+		        xtype: 'textfield',
+		        name: 'qs_user',
+		        reference: 'qs_user_field',
+		        inputId: 'qs_user_field',
+		        value: '',
+		        allowBlank: false,
+		        fieldLabel: gettext('QS Username'),
+		        bind: { hidden: '{!isQuantaStor}' },
+		    }
         ];
 
         me.column2 = [
@@ -67070,7 +67088,7 @@
             {
                 xtype: 'proxmoxcheckbox',
                 name: 'sparse',
-                checked: false,
+                checked: true,
                 uncheckedValue: 0,
                 fieldLabel: gettext('Thin provision'),
             },
@@ -67100,6 +67118,31 @@
                 allowBlank: false,
                 fieldLabel: gettext('Target portal group'),
             },
+            {
+		        xtype: 'proxmoxtextfield',
+		        name: 'qs_apiv4_host',
+		        reference: 'qs_apiv4_host_field',
+		        value: '',
+		        editable: true,
+		        allowBlank: true,
+		        emptyText: Proxmox.Utils.noneText,
+		        bind: { hidden: '{!isQuantaStor}' },
+		        fieldLabel: gettext('QS Host'),
+		        allowBlank: true
+	        },
+	        {
+		        xtype: 'proxmoxtextfield',
+		        name: 'qs_password',
+		        reference: 'qs_password_field',
+		        inputType: me.isCreate ? '' : 'password',
+		        value: '',
+		        editable: true,
+		        deleteEmpty: true,
+		        allowBlank: false,
+		        emptyText: Proxmox.Utils.noneText,
+		        bind: { hidden: '{!isQuantaStor}' },
+		        fieldLabel: gettext('QS Password')
+	        },
         ];
 
         me.callParent();
