--- apidoc.8.4.0.js	2025-11-21 23:21:50.471050049 +0000
+++ apidoc.8.4.0_new.js	2026-01-07 23:59:27.020456325 +0000
@@ -60092,6 +60092,25 @@
                            "type" : "string",
                            "typetext" : "<string>"
                         },
+                        "qs_user" : {
+                           "description" : "QuantaStor user for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "qs_password" : {
+                           "description" : "QuantaStor password for API access",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
+                        "qs_apiv4_host" : {
+                           "description" : "QuantaStor API Host via IPv4",
+                           "format" : "address",
+                           "optional" : 1,
+                           "type" : "string",
+                           "typetext" : "<string>"
+                        },
                         "content" : {
                            "description" : "Allowed content types.\n\nNOTE: the value 'rootdir' is used for Containers, and value 'images' for VMs.\n",
                            "format" : "pve-storage-content-list",
@@ -60627,6 +60646,25 @@
                      "optional" : 1,
                      "type" : "string",
                      "typetext" : "<string>"
+                  },
+                  "qs_user" : {
+                     "description" : "QuantaStor user for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "qs_password" : {
+                     "description" : "QuantaStor password for API access",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
+                  },
+                  "qs_apiv4_host" : {
+                     "description" : "QuantaStor API Host via IPv4",
+                     "format" : "address",
+                     "optional" : 1,
+                     "type" : "string",
+                     "typetext" : "<string>"
                   },
                   "content" : {
                      "description" : "Allowed content types.\n\nNOTE: the value 'rootdir' is used for Containers, and value 'images' for VMs.\n",
